src/Admin/Views/Tools/TaxRate.cshtml

```diff
@@ -0,0 +1,100 @@
+@using Bit.Core.Models.Table
+
+@model TaxRatesModel
+@{
+    ViewData["Title"] = "Tax Rates";
+}
+
+<h1>Manage Tax Rates</h1>
+
+<a class="btn btn-primary mb-2" asp-controller="Tools" asp-action="TaxRateAddEdit">Add a Rate</a>
+<div class="table-responsive">
+    <table class="table table-striped table-hover">
+        <thead>
+            <tr>
+                <th style="width: 190px;">Id</th>
+                <th style="width: 80px;">Country</th>
+                <th style="width: 80px;">State</th>
+                <th style="width: 150px;">Postal Code</th>
+                <th style="width: 160px;">Tax Rate</th>
+                <th style="width: 80px;"></th>
+            </tr>
+        </thead>
+        <tbody>
+            @if(!Model.Items.Any())
+            {
+                <tr>
+                    <td colspan="6">No results to list.</td>
+                </tr>
+            }
+            else
+            {
+                @foreach(var rate in Model.Items)
+                {
+                    <tr>
+                        <td>
+                            @{
+                                var taxRateToEdit = new Dictionary<string, string>
+                                {
+                                    { "id", rate.Id },
+                                    { "stripeTaxRateId", rate.Id },
+                                    { "country", rate.Country },
+                                    { "state", rate.State },
+                                    { "postalCode", rate.PostalCode },
+                                    { "rate", rate.Rate.ToString() }
+                                };
+                            }
+                            <a asp-controller="Tools" asp-action="TaxRateAddEdit" asp-all-route-data="taxRateToEdit">@rate.Id</a>
```

Comment:
Shouldn't you just pass in the rate id and look it up from the database?

Reply:
Done. Note: there is still a separate declaration of "id" and "stripeTaxRateId". This is because the Admin portal has lowercase URLs turned on, so the case-sensitive ID is losing it's value in the URL's Id. Case sensitivity is respected in query string parameters, so I have it in both.